test/test_distributions.py

```diff
@@ -1603,6 +1603,39 @@ def ref_log_prob(idx, x, log_prob):
 
         self._check_log_prob(Normal(loc, scale), ref_log_prob)
 
+    def test_normal_shape(self):
+        for device in torch.testing.get_all_device_types():
+            tensor1 = torch.rand(1, device=device)
+            tensor4 = torch.rand(4, device=device)
+            tensor120 = torch.rand(120, device=device)
+            tensor2145 = torch.rand(2, 1, 4, 5, device=device)
+            tensor2345 = torch.rand(2, 3, 4, 5, device=device)
+            tensor2345_non_contiguous = torch.rand(2, 4, 3, 5, device=device).permute(0, 2, 1, 3)
+            tensor2345_channels_last = tensor2345.contiguous(memory_format=torch.channels_last)
+
+            # inputs have same size
+            self.assertEqual(torch.normal(tensor2345, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345, tensor2345_channels_last).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345_channels_last).size(), (2, 3, 4, 5))
+            # scalar case
+            self.assertEqual(torch.normal(tensor2345, 2).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(2, tensor2345).size(), (2, 3, 4, 5))
+            # inputs are expandable tensors
+            if device == 'cpu':
+                # CPU version is written in legacy code (TH), it doesn't support broadcasting
+                with self.assertRaisesRegex(RuntimeError, "inconsistent tensor size"):
+                    torch.normal(tensor2345, tensor2145)
+            else:
+                self.assertEqual(torch.normal(tensor2345, tensor1).size(), (2, 3, 4, 5))
+                self.assertEqual(torch.normal(tensor2145, tensor2345).size(), (2, 3, 4, 5))
+            # inputs are non-expandable tensors, but they have same number of elements
+            self.assertEqual(torch.normal(tensor120, tensor2345).size(), (120,))
+            self.assertEqual(torch.normal(tensor2345, tensor120).size(), (2, 3, 4, 5))
+            # inputs are non-expandable tensors and they don't have same number of elements
+            with self.assertRaises(RuntimeError):
+                torch.normal(tensor2345, tensor4)
+
```

Comment:
We should add tests for the out= variant. Namely, that if the output tensor's shape does not match the desired shape, then torch.normal should raise an error.

I expect this logic to only exist on CUDA (since the CPU version is using TH and TH didn't care about those semantics...)

Reply:
Added